python/TestHarness/tests/test_QueuePBS.py

```diff
@@ -2,45 +2,117 @@
 
         try:
-            # remove previous session file if exists
+            # remove previous session data and the session file
+            self._clean()
             os.remove(self.pbs_session_file)
         except:
             pass
 
-        self.test_file = os.path.join(os.getenv('MOOSE_DIR'), 'test','tests', 'test_harness', 'queue_job')
-
     def tearDown(self):
         """
         tearDown occurs after every test.
 
-        Clean up old session file. Otherwise, when instancing the
-        QueueManager with the presence of a session file, we trigger
-        a checkStatus stage by default.
-
-        We want to control this manually using mock.
+        Clean up old session file and sub-directories.
         """
         try:
-            # remove session file before each test
+            # remove previous session data and the session file
+            self._clean()
             os.remove(self.pbs_session_file)
         except:
             pass
 
-    def createHarnessAndTesters(self, test_file):
+    def _clean(self):
+        """ clean the session sub-directories only (leave session file intact) """
+        if os.path.exists(self.session_dir):
+            try:
+                shutil.rmtree(self.session_dir)
+            except:
+                raise Exception('Failed to clean up pre-existing session directories')
+
+    @mock.patch.object(util, 'formatResult')
+    def _mockSinglePBSLaunch(self, harness, job, fake_result, mock_result):
         """
-        Return a tuple of the TestHarness, and the list of testers
-        generated by the supplied test_file.
+        Launch 'job' and mock the output of runCommand to 'fake_result', then
+        return a list of the last thing(s) the TestHarness tried to print.
+
+        Note: Because we are wanting to control each launched job, we must
+        mock getNextJobGroup. But by doing so we need to become the Scheduler's
+        job manager per say. Hence why, we are checking and modifying the
+        tester's status.
+        """
+
+        # Limit printing of MagicMock
+        mock_result.return_value = ''
+
+        with mock.patch.object(harness.scheduler, 'getNextJobGroup') as mock_next_jobs:
+            with mock.patch.object(util, 'runCommand') as mock_run:
+                # Setup Mock returns
+                mock_run.return_value = fake_result
+                mock_next_jobs.return_value = []
+                tester = job.getTester()
+
+                if tester.isInitialized():
+                    tester.setStatus('PENDING', tester.bucket_pending)
+
+                # Skipped jobs are sent to a different method and have already
+                # been processed during the iteration that created this
+                # condition in the first place (jobA fails and thus skips and
+                # never runs jobB, but _occurres_ during jobA's iteration).
+                elif tester.isSkipped():
+                    return
+
```

Comment:
This is okay, because the thread timers in play, will handle timeouts/long running and kill this job if something terrible goes wrong.